packages/@uppy/url/src/Url.jsx

```diff
@@ -48,7 +48,9 @@ function checkIfCorrectURL (url) {
 }
 
 function getFileNameFromUrl (url) {
-  return url.substring(url.lastIndexOf('/') + 1)
+  const queryParamsIndexOf = url.indexOf('?')
+  const queryParamsStart = queryParamsIndexOf === -1 ? undefined : queryParamsIndexOf
+  return url.substring(url.lastIndexOf('/', queryParamsStart) + 1, queryParamsStart)
```

Comment:
I think we can use web apis for this
```suggestion
  return new URL(url).origin
```